src/Gizmos/axisDragGizmo.ts

```diff
@@ -176,19 +174,17 @@ export class AxisDragGizmo extends Gizmo {
             this._isHovered = !!(cache.colliderMeshes.indexOf(<Mesh>pointerInfo?.pickInfo?.pickedMesh) != -1);
             if (!this._parent) {
                 const material = this.dragBehavior.enabled ? (this._isHovered || this._dragging ? this._hoverMaterial : this._coloredMaterial) : this._disableMaterial;
-                cache.gizmoMeshes.forEach((m: Mesh) => {
-                    m.material = material;
-                    if ((<LinesMesh>m).color && this.dragBehavior.enabled) {
-                        (<LinesMesh>m).color = material.diffuseColor;
-                    }
-                });
+                this._setGizmoMeshMaterial(cache.gizmoMeshes, material);
             }
         });
+
+        this.dragBehavior.onEnabledObservable.add((newState) => {
+            this._setGizmoMeshMaterial(cache.gizmoMeshes, newState ? cache.material : cache.disableMaterial);
+        });
     }
     protected _attachedNodeChanged(value: Nullable<Node>) {
         if (this.dragBehavior) {
             this.dragBehavior.enabled = value ? true : false;

```

Comment:
Probably a design decision question: would we still need _attachedNodeChanged?
If enabled is something that the developer controls, maybe we no longer need it to be toggled on/off depending on if it's attached to a node or not?